accel-pppd/ctrl/pppoe/pppoe.c

```
@@ -238,23 +233,19 @@ static void ppp_started(struct ap_session *ses)
 	log_ppp_debug("pppoe: ppp started\n");
 }
 
-#ifdef HAVE_VPP
 int pppoe_terminate(struct ap_session *ses, int hard) {
 	int ret = 0;
-	struct ppp_t *ppp = container_of(ses, typeof(*ppp), ses);
-	struct pppoe_conn_t *conn = container_of(ppp, typeof(*conn), ppp);
 
-	if (ppp->is_vpppoe && !conn->serv->is_vpppoe_lost) {
-		vpp_iproute_flush(ses);
-		vpppolicer_remove_limiter(ses);
-		vpppoe_sync_del_pppoe_interface(conn->addr, conn->sid);
-	}
+#ifdef HAVE_SESSION_HOOKS
+	if (ses->hooks && ses->hooks->pppoe_terminate &&
+		ses->hooks->pppoe_terminate(ses, hard))
```

I meant to capture the return value of 'ses->hooks->pppoe_terminate()' and log an error if its an error value (non-zero)

---

I'm not sure what the point of logging error values is. As a part of pppoe plugin, this code is explicitly "not" aware of VPP, so the error value doesn't make any sense IMO.

---

The idea behind is to log some error in case any error occurs. It is more specific to logging 'some' error than the error value. If you feel that there is no harm to have a silent error returned from VPP in this scenario, please feel free to disregard the comment.

---

Well, there is a log, but without the error code value:
```
log_ppp_warn("pppoe: issue to terminate PPPoE session with hook\n");
```

---

Sorry for the confusion, I missed that you have enclosed the function call inside if(). Thank you for putting the log.